### PR TITLE
Remove ledger proposal table and model

### PR DESCRIPTION
**Pull request by kerzhner**
The [latest ledger funding ](https://github.com/statechannels/statechannels/pull/3287) does not use ledger proposals.

Fixes https://github.com/statechannels/statechannels/issues/3371, https://github.com/statechannels/statechannels/issues/3372

### CODE REVIEW

File: packages/server-wallet/src/wallet/wallet.ts [hunk 1]
```diff
@@ -879,7 +862,7 @@ export class SingleThreadedWallet
     response.createdObjectives = storedObjectives;
 
     const channelIdsFromRequests: Bytes32[] = [];
-    const requests = (wirePayload.requests || []).map(deserializeRequest);
+    const requests = wirePayload.requests || [];
```
Comment (geoknee) [inline packages/server-wallet/src/wallet/wallet.ts:865]: Curious about this change? It doesn't seem related to the main thrust of the PR?

File: packages/server-wallet/src/db/migrations/20210312082700_remove_ledger_proposals.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import {
+  up as createLedgerProposalsTable,
+  down as dropLedgerProposalsTable,
+} from './20201201095501_ledger_proposals';
+
+export const up = dropLedgerProposalsTable;
+export const down = createLedgerProposalsTable;
```
Comment (geoknee) [inline packages/server-wallet/src/db/migrations/20210312082700_remove_ledger_proposals.ts:7]: It occurs to me here that there is an opportunity to rewrite history so that this ledger proposals table never existed. What are your thoughts on doing that?

**Review by geoknee — APPROVED** [review]
Looks good to me!


> Fixes #3287, #3372

#3287 is a PR, not an Issue. It's obviously relevant, but I wonder if you meant to link to an issue?

File: packages/server-wallet/src/wallet/wallet.ts (hunk 1 above)
Comment (geoknee) [reply packages/server-wallet/src/wallet/wallet.ts:865]: Nevermind, I see that the `deserializeRequest` function is very specific to ledger proposals, and otherwise is essentially an identity operation. Good call on removing it.

File: packages/server-wallet/src/db/migrations/20210312082700_remove_ledger_proposals.ts (hunk 2 above)
Comment (kerzhner) [reply packages/server-wallet/src/db/migrations/20210312082700_remove_ledger_proposals.ts:7]: I think that it would be nice to collapse all migrations into a single, initial migration since we are no longer on the hook for maintaining/migrating databases that are in use. Filed https://github.com/statechannels/statechannels/issues/3379
